core/src/main/kotlin/behavior/channel/TopGuildChannelBehavior.kt

```diff
@@ -0,0 +1,155 @@
+package dev.kord.core.behavior.channel
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.common.exception.RequestException
+import dev.kord.core.Kord
+import dev.kord.core.cache.data.InviteData
+import dev.kord.core.entity.*
+import dev.kord.core.entity.channel.TopGuildChannel
+import dev.kord.core.exception.EntityNotFoundException
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+import dev.kord.rest.builder.channel.ChannelPermissionModifyBuilder
+import dev.kord.rest.request.RestRequestException
+import dev.kord.rest.service.RestClient
+import dev.kord.rest.service.editMemberPermissions
+import dev.kord.rest.service.editRolePermission
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.withIndex
+import java.util.*
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+/**
+ * The behavior of a Discord channel associated to a [guild].
+ */
+interface TopGuildChannelBehavior : GuildChannelBehavior {
```

Comment:
This should explain what constitutes as a top channel, and how it is different from a not-top channel